optimum/onnxruntime/optimization.py

```diff
@@ -0,0 +1,274 @@
+                Path of a stored onnx model.
+            optimized_model_path (:obj:`str`, `optional`):
+                Path of the corresponding optimized onnx model.
+            summary (:obj:`bool`):
+                Whether report the optimization details: reduction of nodes, and complex node fusions.
+            nodes_details (:obj:`bool`):
+                Whether report the top 5 reduced op_types, and return the detailed node change list.
+        Return:
+            sorted_nodes_change (:obj:`dict`):
+                Returns a sorted list with op types and its change after the optimization.
+        """
+        if self.onnx_model_path is None and onnx_model_path is None:
+            raise AttributeError(
+                "No `self.onnx_model_path` attribute, please input the value of `onnx_model_path` argument."
+            )
+        if self.optim_model_path is None and optimized_model_path is None:
+            raise AttributeError(
+                "No `self.optimized_model_path` attribute, please input the value of `optimized_model_path` argument."
+            )
+        onnx_model_path = onnx_model_path if onnx_model_path else self.onnx_model_path
+        optimized_model_path = optimized_model_path if optimized_model_path else self.optim_model_path
+        onnx_model = load_model(onnx_model_path, format=None, load_external_data=True)
+        optim_model = load_model(optimized_model_path, format=None, load_external_data=True)
+        onnx_model = BertOnnxModel(onnx_model)
+        optimizer = BertOnnxModel(optim_model)
+        
+        def get_node_change(op_type):
+            return len(onnx_model.get_nodes_by_op_type(op_type)) - len(optimizer.get_nodes_by_op_type(op_type))
+
+        if summary:
+            # Nodes reduction information
+            count_nodes_onnx = len(onnx_model.nodes())
+            count_nodes_optim = len(optimizer.nodes())
+            print(f"There are {count_nodes_onnx} nodes before the optimization," 
+                f"and {count_nodes_optim} nodes after the optimization."
+                f" {count_nodes_onnx-count_nodes_optim} nodes are reduced.")
+            # Extended fusion statistics
+            extended_fusion_statistic = optimizer.get_fused_operator_statistics()
+            print(f"Complex node fusions(if extended mode enabled, opt_level>1):\n{extended_fusion_statistic}")
+        
+        # Top 5 reduced operations & node details onnx model v.s. optimized model
+        if nodes_details:
+            op_types = []
+            for model in [onnx_model, optimizer]:
+                for node in model.nodes():
+                    if node.op_type not in op_types: op_types.append(node.op_type)
+                        
+            nodes_change = dict(map(lambda op_type: (op_type, get_node_change(op_type)), op_types))
+            sorted_nodes_change = sorted(nodes_change.items(), key=lambda op: abs(op[1]))
+            sorted_nodes_change.reverse()
```

Comment:
`-abs(op[1]))` would allow to not perform the reverse operation

Reply:
Exact